alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/settings/SupportedApiSettings.kt

```diff
@@ -0,0 +1,66 @@
+package com.amazon.opendistroforelasticsearch.alerting.settings
+
+import com.amazon.opendistroforelasticsearch.alerting.core.model.LocalUriInput
+
+/**
+ * A class that supports storing a unique set of API paths that can be accessed by general users.
+ */
+class SupportedApiSettings {
+    companion object {
+        const val CLUSTER_HEALTH_PATH = "/_cluster/health"
+        const val CLUSTER_STATS_PATH = "/_cluster/stats"
+
+        /**
+         * Each String represents the path to call an API.
+         * NOTE: Paths should conform to the following pattern:
+         * "/_cluster/health"
+         *
+         * Each Set<String> represents the supported JSON payload for the respective API.
+         */
+        private var supportedApiList = HashMap<String, Map<String, Set<String>>>()
+
+        /**
+         * Set to TRUE to enable the supportedApiList check. Set to FALSE to disable.
+         */
+        // TODO: Currently set to TRUE for testing purposes.
+        //  Should likely be set to FALSE by default.
+        private var supportedApiListEnabled = true
+
+        init {
+            supportedApiList[CLUSTER_HEALTH_PATH] = hashMapOf()
+            supportedApiList[CLUSTER_STATS_PATH] = hashMapOf()
+        }
+
+        /**
+         * Returns the map of all supported json payload associated with the provided path from supportedApiList.
+         * @param path The path for the requested API.
+         * @return The map of all supported json payload for the requested API.
+         * @throws IllegalArgumentException When supportedApiList does not contain a value for the provided key.
+         */
+        fun getSupportedJsonPayload(path: String): Map<String, Set<String>> {
+            return supportedApiList[path] ?: throw IllegalArgumentException("API path not in supportedApiList: $path")
+        }
+
+        /**
+         * If [supportedApiListEnabled] is TRUE, calls [validatePath] to confirm whether the provided path
+         * is in supportedApiList. Will otherwise take no actions.
+         * @param localUriInput The [LocalUriInput] to validate.
+         * @return The path that was validated.
+         */
+        fun validateLocalUriInput(localUriInput: LocalUriInput): String {
```

Comment:
Do you think it makes sense to rename this to something like ``resolveToUriPath` as this method seems to be doing more than just validation, its also seems to be converting ``LocalUriInput`` to a path. 

Also do we really need to convert to a uri path? Looks the entity being used later on is transport action request do you think it makes sense to just build and return ActionRequest from this method instead, so callers don't need to worry about building the action request with all the correct attributes?

```
fun resolveToActionRequest(localUriInput: LocalUriInput): ActionRequest? {
  ....
} 
```

And caller can just call this method

Reply:
That makes sense. I'll refactor this method to `resolveToActionRequest` and return the ActionRequest. Thank you for the feedback! I've also made the `validatePath` method public for use in the `Monitor` class.